src/handlers/flowTypeHandler.js

```diff
@@ -17,13 +18,33 @@ import getPropertyName from '../utils/getPropertyName';
 import getFlowTypeFromReactComponent, {
   applyToFlowTypeProperties,
 }  from '../utils/getFlowTypeFromReactComponent';
+import resolveToValue from '../utils/resolveToValue';
+import setPropDescription from '../utils/setPropDescription';
 
+const {types: {namedTypes: types}} = recast;
 function setPropDescriptor(documentation: Documentation, path: NodePath): void {
-  const propDescriptor = documentation.getPropDescriptor(getPropertyName(path));
-  const type = getFlowType(path.get('value'));
-  if (type) {
-    propDescriptor.flowType = type;
+  if (types.ObjectTypeSpreadProperty.check(path.node)) {
+    const name = path.get('argument').get('id').get('name');
+    const resolvedPath = resolveToValue(name);
+
+    if (resolvedPath && types.TypeAlias.check(resolvedPath.node)) {
+      const right = resolvedPath.get('right');
+      applyToFlowTypeProperties(right, propertyPath => {
+        setPropDescriptor(documentation, propertyPath)
+      });
+    } else {
+      documentation.addComposes(name.node.name);
```

Comment:
So this takes the variable name used in the spread syntax, right? Did we plan to use the module name instead at some point? I can't remember.

Reply:
I think where it ended up was that it was too difficult to make something meaningful here.

```
import type { MyType } from 'MyFile';

type MyFileMyProperty = $PropertyType<MyType, 'myProperty'>;

type Props = {|
  ...MyFileMyProperty
|}
```

It doesn't make sense to say that it composes `MyFile.MyFileMyProperty`, in this case it would be `MyFile.MyType.myProperty` but the logic can get exceedingly complicated and it doesn't make sense for us to try to understand it statically. I think where we landed was to just use the alias, if they want it to be more explicit in the generated docs they can use a more explicit alias. 

This still leaves us open to improving it at a future point though.

Reply:
Alright